src/kernel/mreader.ml

```diff
@@ -66,16 +66,23 @@ let instantiate_reader tr spec config source = match spec with
   | name :: args ->
     let reader = lazy (Mreader_extend.start tr name args config source) in
     (reader, (fun () ->
-       if Lazy.is_val reader then
-         match Lazy.force reader with
-         | None -> ()
-         | Some reader -> Mreader_extend.stop tr reader))
+         if Lazy.is_val reader then
+           match Lazy.force reader with
+           | None -> ()
+           | Some reader -> Mreader_extend.stop tr reader))
 
 let get_reader config =
+  let rec find_reader assocsuffixes =
+    match assocsuffixes with
+    | [] when Filename.check_suffix Mconfig.(config.query.filename) ".re"
+           || Filename.check_suffix Mconfig.(config.query.filename) ".rei"  -> ["reason"]  
```

Comment:
I'd vote for pre-populating the `assocsuffix` field of `Mconfig.initial` with `["reason"]`, instead of keeping that special case here.

Reply:
ok good idea - will implement this